src/main/java/com/aws/greengrass/secretmanager/SecretManager.java

```diff
@@ -207,9 +238,10 @@ public void loadSecretsFromLocalStore() throws SecretManagerException {
     * arn1:l1 -> secret3
     * arn1:l2 -> secret4
     */
-    private void loadCache(AWSSecretResponse awsSecretResponse) throws SecretManagerException {
+    private void loadCache(AWSSecretResponse awsSecretResponse) throws SecretManagerException, InterruptedException {
         GetSecretValueResponse decryptedResponse = null;
         try {
+            waitForInitialization();
```

Comment:
What are the default timeouts for IPC API's. What happens when getSecret API is called before the security service comes up?

Reply:
There is no timeout; that's up to the customer how long they will wait

Reply:
What happens is that we will be waiting until the service becomes available.